pyirf/irf/energy_dispersion.py

```diff
@@ -110,49 +110,55 @@ def energy_dispersion_to_migration(
     new_reco_energy_edges: astropy.units.Quantity[energy]
         Reco energy edges matching the second dimension of the output
 
-    Returns:
-    --------
+    Returns
+    -------
     migration_matrix: numpy.ndarray
         Three-dimensional energy migration matrix. The third dimension
         equals the fov offset dimension of the energy dispersion matrix.
     """
-
-    migration_matrix = np.zeros((
-        len(new_true_energy_edges) - 1,
-        len(new_reco_energy_edges) - 1,
-        dispersion_matrix.shape[2],
-    ))
+    migration_matrix = np.zeros(
+        (
+            len(new_true_energy_edges) - 1,
+            len(new_reco_energy_edges) - 1,
+            dispersion_matrix.shape[2],
+        )
+    )
 
     true_energy_interpolation = resample_histogram1d(
         dispersion_matrix,
         disp_true_energy_edges,
         new_true_energy_edges,
         axis=0,
+        normalize=True
```

Comment:
I think you should transform the dispersion_matrix from a pdf to the probabilities before calling `resample_histogram1d` and then the rest of the code can stay the same.

I.e.:
```
resample_histogram1d(
    dispersion_matrix / np.diff(disp_migration_edges)[np.newaxis, :, np.newaxis],
...
)

Reply:
but you still need to normalize the final migration matrix
if you do that + what you propose you obtain the same result I show here

(also sorry that `normalize=True` part is some debugging code I added, I'll remove it)

Reply:
I squashed the removal of the debugging code and addition of the news fragment, you should re-pull if you pulled before.

Reply:
> but you still need to normalize the final migration matrix

No, why? The interpolation should produce the correct normalization.

Reply:
oh wait I think this is because in my code the original matrix was normalized so the sum is 1, not the PDF

in the case of the integral = 1 the peaks can still be greater than 1 so I guess its OK but I guess it's better to add to the test the check that the area is 1

Reply:
However if one plots the same matrix using gammapy from a kernel (provided we account for the fact that in that case you have to choose an FoV offset) one gets a z-axis which goes up to 1 so it should be normalized with the sum=1

Reply:
I think what I did is correct though: the migration matrix should be normalized as having probabilities along the e_reco axis and I checked that the sum along that axis is 1 for each e_true

if I do not normalize the final matrix after dividing by the migration bin widths this is not true anymore

Reply:
If you handle the input EDISP correctly, not renormalization should be required.

In the first comment in this thread I posted how I think it should be done. Could you please try?

Reply:
yeah I did that and it didn't work (also it's wrong, should be a multiplication)